sphere-models/src/it/java/io/sphere/sdk/customers/commands/CustomerSignInCommandTest.java

```diff
@@ -21,6 +38,53 @@ public void execution() throws Exception {
     }
 
     @Test
+    public void signInWithAnonymousCart() throws Exception {
+        withCustomerCustomerCartAndAnonymousCart(client(), customer -> customersCart -> anonymousCart -> {
+            assertThat(customersCart.getCustomerId())
+                    .as("customersCart belongs to customer")
+                    .isEqualTo(customer.getId());
+            assertThat(anonymousCart.getCustomerId()).as("anonymous cart has no customer").isNull();
+            assertThat(customersCart.getLineItems().get(0).getProductId())
+                    .as("both carts have the same product variant")
+                    .isEqualTo(anonymousCart.getLineItems().get(0).getProductId());
+            assertThat(customersCart.getLineItems().get(0).getQuantity())
+                    .describedAs("both carts have the same variant in different quantities")
+                    .isNotEqualTo(anonymousCart.getLineItems().get(0).getQuantity());
+            assertThat(customersCart.getLineItems().get(0).getQuantity()).isEqualTo(3);
+            assertThat(anonymousCart.getLineItems().get(0).getQuantity()).isEqualTo(7);
+            final long maxOfBothQuantities = Math.max(customersCart.getLineItems().get(0).getQuantity(),
+                    anonymousCart.getLineItems().get(0).getQuantity());
+            assertThat(maxOfBothQuantities).isEqualTo(7);
+
+            final CustomerSignInResult result = execute(CustomerSignInCommand
+                    .of(customer.getEmail(), PASSWORD, anonymousCart.getId()));
+
+            final Cart mergeResultCart = result.getCart();
+            final LineItem mergeResultLineItem = mergeResultCart.getLineItems().get(0);
+            assertThat(mergeResultLineItem.getQuantity()).isEqualTo(maxOfBothQuantities).isEqualTo(7);
+            assertThat(mergeResultLineItem.getProductId())
```

Comment:
Really? I have always expected it to be the sum of both: 10. Although I don't have any strong reason against/pro any form. Do you know why the maximum quantity?